packages/server/src/core/internals/procedureBuilder.ts

```diff
@@ -192,18 +192,19 @@ function createNewBuilder(
   def1: AnyProcedureBuilderDef,
   def2: Partial<AnyProcedureBuilderDef>,
 ) {
-  const { middlewares = [], inputs, ...rest } = def2;
+  const { middlewares = [], inputs, meta, ...rest } = def2;
 
   // TODO: maybe have a fn here to warn about calls
   return createBuilder({
     ...mergeWithoutOverrides(def1, rest),
     inputs: [...def1.inputs, ...(inputs ?? [])],
     middlewares: [...def1.middlewares, ...middlewares],
-  } as any);
+    meta: def1.meta && meta ? { ...def1.meta, ...meta } : meta ?? def1.meta,
```

Comment:
```suggestion
    meta: { ...def1.meta, ...meta }
```

It's safe to spread possibly-undefined objects - https://github.com/tc39/ecma262/issues/687
And the readability of this version outweighs creating a "not always necessary" new object for each builder IMO.

In case

Reply:
Great suggestion, that is much cleaner yes!

For L195 the purpose of that is actually to separate keys we want to merge manually from keys we want to merge automatically (`..rest`) so that `mergeWithoutOverrides` can throw for any duplicates, so I think that's best as it is

Reply:
Had to revert off this change because it's actually a breaking change, `undefined` is a valid meta value and middlewares in the wild might check for this. Tests started failing

Reply:
Oh, I see. Quite a bummer. As for meta merging, we can then change it to this:

```js
meta: meta ? { ...def1.meta, ...meta } : def1.meta
```

```js
const merge = (a, b) => b.meta ? { ...a.meta, ...b.meta } : a.meta;

console.log(merge({}, {})); // undefined
console.log(merge({ meta: {a:1} }, { meta: undefined })); // {a:1}
console.log(merge({ meta: undefined }, { meta: {b:1} })); // {b:1}
console.log(merge({ meta: {a:1} }, { meta: {a:2} })); // {a:2}
console.log(merge({ meta: {a:1} }, { meta: {b:1} })); // {a:1, b:1}
```

Reply:
Unless `def2.meta` can also be `undefined`, then we're out of luck ¯\_(ツ)_/¯

Reply:
Yep either could be undefined. Could shunt this into a function though?

Reply:
Hmm, if `def2.meta` can be `undefined`, does this mean that it overrides the `def1` entirely or leave It as is?
Because if leave as is, then it'll work just fine with the example above, if it needs to override it, then your solution is probably the best one we can get. I think its fine without separate function, just wrote it like this for easier testing :)

Reply:
It's a good question. 

My understanding is:
* Both can be undefined (no meta set, and override called for other stuff)
* def2 is necessarily partial, def2.meta will not be passed if the override call is for some other key(s), and def1.meta should be retained
* Users can't set `procedure.meta(undefined)` if `initTRPC.meta<>()` has been called with a type, so TS prevents unexpected behaviour as a user
* def1.meta could be undefined and def2.meta could have a value if `initTRPC.meta<>()` hasn't been called

So the way I wrote this is necessarily safe, because basically every combination is a possibility, but the precedence is always "if both then merge otherwise return the non-nully one"

Reply:
> "if both then merge otherwise return the non-nully one"

`(def1.meta || meta) ? { ...def1.meta, ...meta } : undefined`

Then this should do as well, but it's basically stylistic at this point.

Reply:
Yeah I do agree we're in the stylistic weeds now. I can't think of any scenario where meta would be something other than an object or undefined, so in principle this is fine, but my instinct is to outsource the value decision upward and to honour whatever we're given if merging isn't needed - if that's `null` for some reason this function shouldn't care.

Happy to apply the change if you like :)

Reply:
Yeah, let's leave it like this then :)